### PR TITLE
Log request data alongside pipeline id for traceability

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy [hunk 1]
```diff
@@ -363,24 +365,18 @@ class OperationsController {
...
-  private Map<String, String> markPipelineFailed(Map config, Exception e) {
+  private String markPipelineFailed(Map config, Exception e) {
     injectPipelineOrigin(config)
     def json = objectMapper.writeValueAsString(config)
-    log.warn('requested pipeline marked as failed: {}', json)
-
     def pipeline = executionLauncher.fail(PIPELINE, json, e)
-
-    [ref: "/pipelines/${pipeline.id}".toString()]
+    return pipeline.id
```
Comment (robfletcher) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: The previous return value from this method was being ignored as the calling code immediately throws an exception.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy [hunk 2]
```diff
@@ -363,24 +365,18 @@ class OperationsController {
...
-  private Map<String, String> startPipeline(Map config) {
+  private String startPipeline(Map config) {
     injectPipelineOrigin(config)
     def json = objectMapper.writeValueAsString(config)
-    log.info('requested pipeline: {}', json)
-
     def pipeline = executionLauncher.start(PIPELINE, json)
-
-    [ref: "/pipelines/${pipeline.id}".toString()]
+    return pipeline.id
```
Comment (robfletcher) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: Modified here to just return the id so I can log it.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy [hunk 3]
```diff
@@ -183,9 +188,6 @@ class OperationsController {
       pipeline = preprocessor.process(pipeline)
     }
 
-    def json = objectMapper.writeValueAsString(pipeline)
-    log.info('received pipeline {}:{}', value("pipelineId", pipeline.id), json)
```
Comment (robfletcher) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: This logging seems to be just noise.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy [hunk 4]
```diff
@@ -150,6 +150,8 @@ class OperationsController {
   }
 
   private Map<String, Object> orchestratePipeline(Map pipeline) {
+    def request = objectMapper.writeValueAsString(pipeline)
```
Comment (robfletcher) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: Grab the original, un-fucked-with request.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy [hunk 5]
```diff
@@ -165,9 +167,12 @@ class OperationsController {
     processedPipeline.trigger = objectMapper.convertValue(processedPipeline.trigger, Trigger)
 
     if (pipelineError == null) {
-      startPipeline(processedPipeline)
+      def id = startPipeline(processedPipeline)
+      log.info("Started pipeline {} based on request body {}", id, request)
```
Comment (robfletcher) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: Log out the id and original request JSON.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy (hunk 3 above)
Comment (emjburns) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy]: Are you "replacing" this with the logs that either are 'started' or 'failed'?

**Pull request merged**